hantec_api_ecommerce/controllers/main.py

```diff
@@ -0,0 +1,570 @@
+
+from odoo.http import request, Response, Controller, route
+import json
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+class MainController(Controller):
+    @route('/create_contact', methods=['POST'], type='json', auth='user' )
+    def create_contact(self):
+        
+        # Lista de todos los campos posibles
+
+        all_fields = [
+            'company_type', 'name', 'phone', 'street_number', 'street_number2',
+            'email', 'street', 'street2', 'city', 'zip', 'state_id', 'country_id',
+            'vat', 'l10n_mx_edi_curp', 'website', 'lang', 'category_id', 'l10n_mx_edi_fiscal_regime', 'parent_id'
+        ]
+
+        # Construir el diccionario contact_data con solo los campos enviados en el JSON
+        contact_data = {field: request.jsonrequest.get(field) for field in all_fields if request.jsonrequest.get(field) is not None}
```

Comment:
```suggestion
        contact_data = {
            field: request.jsonrequest.get(field) for field in all_fields if request.jsonrequest.get(field) is not None
        }
```

take care of indentation please

you can use black 
https://pypi.org/project/black/